packages/block-editor/src/components/block-list/block.native.js

```diff
@@ -170,35 +230,48 @@ export default compose( [
 			getBlockIndex,
 			getBlocks,
 			isBlockSelected,
-			__unstableGetBlockWithoutInnerBlocks,
 			getBlockHierarchyRootClientId,
+			getSelectedBlockClientId,
 			getBlock,
 			getBlockRootClientId,
-			getSelectedBlock,
+			getLowestCommonAncestorWithSelectedBlock,
+			getBlockParents,
 		} = select( 'core/block-editor' );
 		const order = getBlockIndex( clientId, rootClientId );
 		const isSelected = isBlockSelected( clientId );
-		const isFirstBlock = order === 0;
 		const isLastBlock = order === getBlocks().length - 1;
-		const block = __unstableGetBlockWithoutInnerBlocks( clientId );
+		const block = getBlock( clientId );
```

Comment:
I have concerns about changing this one. Because I use it to check if block `hasChildren` but I'm not sure if there is another way to check that ?

Reply:
If you don't need the block for anything else, you can also use `getBlockCount( clientId )` to get the number of children

Reply:
thanks ! I have used this approach in latest changes